File: Core/Models/Entities/Product.cs
using System.ComponentModel;

namespace Core.Entities;

public class Product : BaseEntitie
{
    
    public string Name  { get; set; }
    
    public string Description { get; set; }
    
    public int Stock { get; set; }
    
    public double AverageRating { get; set; }

    public Categorie Categorie { get; set; }
    
    public int? CategoryID { get; set; } 
    
    public ICollection<OrderDetail>  OrdersDetails {get; set;}
    public ICollection<ProductPrice>  ProductPrices {get; set;}
    
    public ICollection<CartItem> CartItems { get; set; } 
    
    public TeaDetail? TeaDetail { get; set; }
    
    public ICollection<Review>  Reviews {get; set;}
    public ICollection<Photo>  Photos {get; set;}
    


}

File: client/src/app/models/product.ts
export interface Product {
  id: number,
  name: string,
  description: string,
  stock: number,
  categoryID: number | null,
  rating: number | null,
  productPrices?: ProductPrice[];  // Добавлено
  photos?: Photo[];                 // Добавлено
  teaDetail?: TeaDetail;
}

export interface TeaDetail {
  id: number,
  history: string;
  preparationGuide: string;
  tastingNotes: string;
  origin: string;
  storageInstruction: string;
}

export interface Photo {
  image?: string;
}

export interface ProductPrice {
  id: number,
  price: number,
  weightGrams: number;
}
